src/shared/lib/fetch/index.js

```diff
@@ -162,17 +162,29 @@ export const pdf = async (scraper, url, cacheKey = 'default', date, options) =>
   return data;
 };
 
-const fetchHeadless = async url => {
+/**
+ * Load a url in headless browser, call user-supplied callback, return callback's output
+ * @param {*} scraper the scraper object
+ * @param {string} url URL of the resource
+ * @param {null} cacheKey The cache key -- this is IGNORED by this function as data is dynamically extrated and cannot be cached
+ * @param {function} callback The callback to be used on open page. Should accept a single argument `page` that is a puppeteer page that has been loaded with the given url
+ */
+export const fetchHeadlessCallback = async (scraper, url, callback, cacheKey = null) => {
+  if (cacheKey !== null) {
+    throw new Error(
+      'cacheKey is ignored in fetchHeadlessCallback as data is dynamically extracted by callback each time the function is called'
```

Comment:
Even though it's dynamic, I think we should still cache what we can.  At one point in the callback, we have a page (which you parse) which we can likely cache.  I'm taking a look at that.

Reply:
Also, if the method should never be passed a param, the method shouldn't have the param in the signature.